sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -295,6 +307,8 @@ private int readPageV1(DataPageV1 page) throws IOException {
   }
 
   private int readPageV2(DataPageV2 page) throws IOException {
+    this.pageFirstRowIndex = page.getFirstRowIndex().orElse(0L);
```

Comment:
Maybe move to `readPage`? Looks like `readPageV1` and `readPageV2` all need it.

Reply:
Good point. Let me do that.